packages/builder/src/pages/builder/app/[application]/_layout.svelte

```diff
@@ -102,6 +107,15 @@
     }
   }
 
+  const showFreeTrialModal = (planType, freeTrialModal) => {
+    if (
+      planType === "enterprise_basic_trial" &&
+      !$auth.user?.freeTrialConfirmedAt
```

Comment:
Will this not spam the modal until confirmed?

Reply:
The modal has this property: `disableCancel={true}` which means the user must confirm the modal before they can interact with the app.

With that said, if the API call fails for any reason, you would really want the modal to close and stay closed. Might just put in an additional local variable to account for that scenario

Reply:
Although `$: showFreeTrialModal(planType, freeTrialModal)` will only trigger initially, so I think it's alright to stay as is.